gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/BigqueryRpc.java

```diff
@@ -0,0 +1,142 @@
+import com.google.api.services.bigquery.model.Table;
+import com.google.api.services.bigquery.model.TableDataInsertAllRequest;
+import com.google.api.services.bigquery.model.TableDataInsertAllResponse;
+import com.google.api.services.bigquery.model.TableReference;
+import com.google.api.services.bigquery.model.TableRow;
+import com.google.gcloud.bigquery.BigqueryException;
+
+import java.util.Map;
+
+public interface BigqueryRpc {
+
+  // These options are part of the Google Cloud BigQuery query parameters
+  enum Option {
+    QUOTA_USER("quotaUser"),
+    USER_IP("userIp"),
+    FIELDS("fields"),
+    DELETE_CONTENTS("deleteContents"),
+    ALL_DATASETS("all"),
+    ALL_USERS("allUsers"),
+    MAX_RESULTS("maxResults"),
+    PAGE_TOKEN("pageToken"),
+    START_INDEX("startIndex"),
+    STATE_FILTER("stateFilter"),
+    TIMEOUT("timeOut");
+
+    private final String value;
+
+    Option(String value) {
+      this.value = value;
+    }
+
+    public String value() {
+      return value;
+    }
+
+    @SuppressWarnings("unchecked")
+    <T> T get(Map<Option, ?> options) {
+      return (T) options.get(this);
+    }
+
+    String getString(Map<Option, ?> options) {
+      return get(options);
+    }
+
+    Long getLong(Map<Option, ?> options) {
+      return get(options);
+    }
+
+    Boolean getBoolean(Map<Option, ?> options) {
+      return get(options);
+    }
+  }
+
+  class Tuple<X, Y> {
+
+    private final X x;
+    private final Y y;
+
+    private Tuple(X x, Y y) {
+      this.x = x;
+      this.y = y;
+    }
+
+    public static <X, Y> Tuple<X, Y> of(X x, Y y) {
+      return new Tuple<>(x, y);
+    }
+
+    public X x() {
+      return x;
+    }
+
+    public Y y() {
+      return y;
+    }
+  }
+
+  Dataset getDataset(String datasetId, Map<Option, ?> options) throws BigqueryException;
+
+  Tuple<String, Iterable<Dataset>> listDatasets(Map<Option, ?> options) throws BigqueryException;
+
+  Dataset create(Dataset dataset, Map<Option, ?> options) throws BigqueryException;
+
+  boolean deleteDataset(String datasetId, Map<Option, ?> options) throws BigqueryException;
+
+  Dataset patch(Dataset dataset, Map<Option, ?> options) throws BigqueryException;
+
+  Table getTable(String datasetId, String tableId, Map<Option, ?> options) throws BigqueryException;
+
+  Tuple<String, Iterable<Table>> listTables(String dataset, Map<Option, ?> options)
+      throws BigqueryException;
+
+  Table create(String dataset, Table table, Map<Option, ?> options) throws BigqueryException;
+
+  boolean deleteTable(String datasetId, String tableId, Map<Option, ?> options)
+      throws BigqueryException;
+
+  Table patch(Table table, Map<Option, ?> options) throws BigqueryException;
+
+  TableDataInsertAllResponse insertAll(TableReference table, TableDataInsertAllRequest request,
+      Map<Option, ?> options) throws BigqueryException;
+
+  Tuple<String, Iterable<TableRow>> listTableData(String datasetId, String tableId,
```

Comment:
Well `kind` is the type of response (`bigquery#tableDataList`) and it's usually there in the APIs but we never used it.

`etag` might be used to check the sanity of the response but since the list is paginated we are most likely going to use `Page` which does not expose any `etag` information.